lib/enum/arrays.rb

```diff
@@ -1,24 +1,86 @@
 class Array
+  def iniinitialize
+    @test_array
+  end
 
   def sum_of_numeric_elements
-  end
+    self.map{|num|
+      if num.is_a? (Numeric)
+        num.to_f
+      else
+        0.00
+      end
+    }.compact.reduce(:+)
+    end
 
   def product_of_numeric_elements
+    self.select{|num| num.is_a? (Numeric)}.inject(:*)
+
   end
 
   def even_numeric_elements
+    (self.select{|num| num.is_a? (Numeric)}).map{
+        |new_num| new_num.round(0)}.map{|fun|
+          if fun.even?
+            fun
+          else
+            nil
+          end}.compact
```

Comment:
Couple things about syntax here. 

1. You don't need parentheses at the beginning. You can just do `self.select { ... }.map { ... }`

2. It seems like you're trying to name your local variables to reflect the fact that they are different each time. That's a valid strategy while you're in the middle of writing the method, but when you're done and it works, it's a good idea to change them back to what they represent. (Fun doesn't make a whole lot of sense here, though I see what you're trying to do with it.) 

Again, you make this a whole lot shorter, like this: 

`self.select { |num| num.is_a? Numeric }.select { |num| num.even? }`